compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/symtab_transformer/transformer.rs

```diff
@@ -0,0 +1,742 @@
+    }
+
+    /// Transforms a return stmt (`return e;` or `return;`)
+    fn transform_return_stmt(&self, value: &Option<Expr>) -> Stmt {
+        let transformed_value = value.as_ref().map(|value| self.transform_expr(value));
+        Stmt::ret(transformed_value, Location::none())
+    }
+
+    /// Transforms a skip stmt (`;`)
+    fn transform_skip_stmt(&self) -> Stmt {
+        Stmt::skip(Location::none())
+    }
+
+    /// Transforms a switch stmt (`switch (control) { case1.case: cast1.body; case2.case: case2.body; ... }`)
+    fn transform_switch_stmt(
+        &self,
+        control: &Expr,
+        cases: &[SwitchCase],
+        default: &Option<Stmt>,
+    ) -> Stmt {
+        let transformed_control = self.transform_expr(control);
+        let transformed_cases = cases
+            .iter()
+            .map(|case| {
+                SwitchCase::new(self.transform_expr(case.case()), self.transform_stmt(case.body()))
+            })
+            .collect();
+        let transformed_default = default.as_ref().map(|default| self.transform_stmt(default));
+
+        Stmt::switch(transformed_control, transformed_cases, transformed_default, Location::none())
+    }
+
+    /// Transforms a while loop stmt (`while (cond) { body }`)
+    fn transform_while_stmt(&self, cond: &Expr, body: &Stmt) -> Stmt {
+        let transformed_cond = self.transform_expr(cond);
+        let transformed_body = self.transform_stmt(body);
+        Stmt::while_loop(transformed_cond, transformed_body, Location::none())
+    }
+
+    /// Transforms a symbol's type and value
+    fn transform_symbol(&self, symbol: &Symbol) -> Symbol {
+        let new_typ = self.transform_type(&symbol.typ);
+        let new_value = self.transform_value(&symbol.value);
+        let mut new_symbol = symbol.clone();
+        new_symbol.value = new_value;
+        new_symbol.typ = new_typ;
+        new_symbol
+    }
+
+    /// Transforms a symbol value
+    fn transform_value(&self, value: &SymbolValues) -> SymbolValues {
+        match value {
+            SymbolValues::None => SymbolValues::None,
+            SymbolValues::Expr(expr) => SymbolValues::Expr(self.transform_expr(expr)),
+            SymbolValues::Stmt(stmt) => SymbolValues::Stmt(self.transform_stmt(stmt)),
+        }
+    }
+
+    /// Preprocessing to perform before adding transformed symbols
+    fn preprocess(&mut self) {}
+
+    /// Postprocessing to perform after adding transformed symbols
+    fn postprocess(&mut self) {}
+
+    /// Transforms the orig_symtab, producing a new one.
+    /// See `Transformer` trait documentation for details.
+    fn transform_symbol_table(mut self, orig_symtab: &SymbolTable) -> SymbolTable {
+        self.preprocess();
+
+        let mut added: HashSet<String> = HashSet::new();
+
+        for (name, _symbol) in self.symbol_table().iter() {
+            added.insert(name.clone());
+        }
+
+        for (name, symbol) in orig_symtab.iter() {
```

Comment:
Comment why this separate loop for None is needed.

Reply:
Fixed!